src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,170 @@
+     *
+     * @param input The given input string from .txt file.
+     * @return A task based on the given input string.
+     * @throws DukeException If parse fails.
+     */
+    public static Task parseDocument(String input) {
+        String[] array = input.split(SPLITER);
+        switch (array[0]) {
+        case "T":
+            return new Todo(array[2], parseIsDone(array[1]));
+        case "E":
+            return new Event(array[2], parseIsDone(array[1]), parseDateTime(array[3]));
+        case "D":
+            return new Deadline(array[2], parseIsDone(array[1]), parseDateTime(array[3]));
+        default:
+            throw new DukeException("Error in saved tasks document");
+        }
+    }
+
+    /**
+     * Returns LocalDateTime instance based on the input string.
+     *
+     * @param input The given input string whose format follows dd/MM/yyyy HH:mm.
+     * @return A LocalDateTime instance based on the input string.
+     */
+    public static LocalDateTime parseDateTime(String input) {
+        try {
+            LocalDateTime ldt = LocalDateTime.parse(input, INPUT_DATE_TIME_FORMATTER);
+            return ldt;
+        } catch (DateTimeParseException e) {
+            throw new DukeException("time should be in the format: DD/MM/YYYY HH:MM");
+        }
+    }
+
+    /**
+     * Returns a command based on the given input.
+     *
+     * @param input The given input.
+     * @return A command.
+     * @throws DukeException When parse exception happens.
+     */
+    public static Command parseCommand(String input) {
+        int firstWordIndex = input.indexOf(" ");
+        String actionText = firstWordIndex == -1 ? input : input.substring(0, firstWordIndex);
+        Action action = Parser.getAction(actionText);
+        String rest = input.substring(firstWordIndex + 1);
+        switch (action) {
+        case BYE: {
+            return new ExitCommand(Action.BYE);
+        }
+        case LIST: {
+            return new ShowCommand(Action.LIST);
+        }
+        case DONE: {
+            try {
+                int taskNumber = Integer.parseInt(rest);
+                return new SetCommand(Action.DONE, taskNumber - 1);
+            } catch (NumberFormatException e) {
+                throw new DukeException("A number must be given to specified the task.");
+            }
+        }
+        case TODO: {
+            if (firstWordIndex == -1) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            }
+            return new AddCommand(Action.TODO, rest);
+        }
+        case DEADLINE: {
+            if (firstWordIndex == -1) {
+                throw new DukeException("The description of a deadline cannot be empty.");
+            }
+            return new AddCommand(Action.DEADLINE, rest);
+        }
+        case EVENT: {
+            if (firstWordIndex == -1) {
+                throw new DukeException("The description of an event cannot be empty.");
```

Comment:
Maybe you want to consider abstracting out the process of throwing the DukeException for the description of a certain task being empty? As seen above, all the cases look for whether firstWordIndex's value is -1, and in which case it'd throw the appropriate exception where the string only differs by the type of task!